python/cudf/CMakeLists.txt

```diff
@@ -62,6 +78,14 @@ if(NOT cudf_FOUND)
   install(TARGETS cudf DESTINATION ${cython_lib_dir})
 endif()
 
+if(USE_LIBARROW_FROM_PYARROW)
+  # TODO: It would be nice to avoid setting these globally, but we'd need to manually set this for
+  # each of the rapids-cython targets otherwise which seems undesirable. For now I'm setting this
+  # after the `add_subdirectory(cpp)` call to limit the scope, maybe that's good enough.
```

Comment:
Can you think of any cleaner way to set up these variables rather than modifying the `CMAKE_*_FLAGS` variables directly? I think it might be OK here since it should only affect things compiled below this which are just our own extension modules, it just feels like a slightly more extensive modification than I would hope.

Reply:
Yeah we should be able to attach these as INTERFACE target_compile_options to the `${ARROW_LIBRARIES}` the same we do for include directories. That will isolate the flag to only targets that link to arrow.

Reply:
Good call, this is much nicer.